synapse/cores/ram.py

```diff
@@ -14,9 +14,27 @@ def _initCortex(self):
         self.initSizeBy('le',self._sizeByLe)
         self.initRowsBy('le',self._rowsByLe)
 
+        self.initSizeBy('in',self._sizeByIn)
+        self.initRowsBy('in',self._rowsByIn)
+
         self.initSizeBy('range',self._sizeByRange)
         self.initRowsBy('range',self._rowsByRange)
 
+    def _sizeByIn(self, prop, valu, limit=None):
+        # HACK: for speed
+        data = dict(size=0)
+        def inc():
+            data['size'] += 1
+        [ inc() for r in self.rowsbyprop.get(prop,()) if r[2] in valu ]
+        return data['size']
+
+    def _rowsByIn(self, prop, valu, limit=None):
+        # HACK: for speed
+        ret = [ r for r in self.rowsbyprop.get(prop,()) if r[2] in valu ]
+        if limit != None:
+            ret = ret[:limit]
+        return ret
+
```

Comment:
Looks like we probably need a default set of sizeByIn/rowsByIn methods that just iterate over the given values and ask with standard getFooByProp. Ideally, we could work up a set of db optimized versions provided that all the values in the list are one primitive type (int/str) to actually issue "in" queries to the underlying db instances.

That way, the "in" handlers could be registered by the base cortex constructor and method overrides would provide optimizations.